cmd/clusterctl/validation/validate_pods.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type validationError struct {
+	name    string
+	message string
+}
+
+func ValidatePods(w io.Writer, c client.Client, namespace string) error {
```

Comment:
Shouldn't this function take a context argument? The other functions that it calls use `context.TODO()` but that's more typical for existing code where context has yet to be plumbed through. This is new code, and I don't see any reason it shouldn't pass down context from the start.

Reply:
am I right doing it like this? https://github.com/kubernetes-sigs/cluster-api/pull/679/commits/0123f104fe86be502d65326ed971e1fb5133dcc8#diff-6334f0ae8a2b2bebd8b635352f2a9426R81

I was just trying to make it consistent with the other validation function `ValidateClusterAPIObjects`, is this also opportunity to fix passing context here? if so, I'll address in another PR once this one gets merged.
https://github.com/kubernetes-sigs/cluster-api/blob/master/cmd/clusterctl/validation/validate_cluster_api_objects.go#L56
https://github.com/kubernetes-sigs/cluster-api/blob/master/cmd/clusterctl/validation/validate_cluster_api_objects.go#L46
https://github.com/kubernetes-sigs/cluster-api/blob/master/cmd/clusterctl/validation/validate_cluster_api_objects.go#L61

Reply:
Yes, that looks better. Fixing the others in a follow-up PR sounds great.